### ISSUE TITLE
[Bug] Connecting or removing a Bluetooth causes the player to crash

### ISSUE THREAD

**Issue by larsk2009** [SuRGeoNix/Flyleaf#41]
**Describe the bug**
When playing a video and switching to/from a bluetooth device, the player stops playing the video and the application hangs.

**To Reproduce**
1. Open the WPF sample
2. Play a video
3. Connect/disconnect a bluetooth audio device
4. The player should freeze

**Expected behavior**
The player doesn't freeze and the audio is transferred to or from the bluetooth audio device.

**Additional context**
I have determined the issue is most likely caused by this line:
https://github.com/SuRGeoNix/Flyleaf/blob/05c307b9f5898d077e5aee2a2a31c1b0ae7349ac/FlyleafLib/MediaPlayer/AudioPlayer.cs#L131
That line hangs when connecting or disconnecting a bluetooth device, it seems like there might be a deadlock in there somewhere. 
If I comment that line out, the issue disappears completely. I have also tried to put a `lock` around it, which seems to _sometimes_ fix the issue when connecting the bluetooth device, but never when disconnecting. I have also looked on the internet and there it is also suggested this might be a threading/deadlock issue, so the solution should probably be searched for in that area.

I will do some investigation myself to see if I can fix this issue, but if anyone has any ideas let me know.

**Comment by SuRGeoNix** [comment]
Hey @larsk2009, well nice research so far. Unfortunately, I know this kind of issue... I came across a lot times with NAudio. It happens when you initialize a NAudio player from one thread and dispose it from another. I was able to  solve the most issue by switching to direct sound and creating the AudioMaster.cs to handle the device removal events etc. So, I guess the difference with the bluetooth device is that I'm missing/not handling properly an event on AudioMaster. I will need to find a bluetooth device for testing it tho.

**Comment by larsk2009** [comment]
Yeah I did check the events in AudioMaster.cs, and the `OnDefaultDeviceChanged` and `OnDeviceRemoved` events do get fired when the bluetooth device gets connected or disconnected. I do agree it seems like a threading issue.

To make matters worse, I actually found with more testing that the issue happens every time when disconnecting the device, but only sometimes when connecting it.

**Comment by SuRGeoNix** [comment]
Hey, as I don't have a bluetooth adapter and device to test if you want test if initializing and disposing happens at the same thread... and if that's the problem.

AudioPlayer.cs:Initialize()
```
Log($"AudioPlayer initializing on thread {System.Threading.Thread.CurrentThread.ManagedThreadId}");
```

AudioPlayer.cs:Dispose()
```
if (player != null) Log($"AudioPlayer disposing on thread {System.Threading.Thread.CurrentThread.ManagedThreadId}");
```

I might have to create a thread only for all the naudio players for initializing/playing/disposing at the same thread. I'm afraid that it causes other issues as well. By looking at naudio's issues you will see a lot similar. Seems to be an issue with COM objects.

**Comment by SuRGeoNix** [comment]
Should be fixed in v3.4.2. I wasn't able to test it so let me know if still not working!
